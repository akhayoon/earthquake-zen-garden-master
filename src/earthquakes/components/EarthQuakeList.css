.earthquake-list {
  border-collapse: collapse;
}

.earthquake-list tr th {
  cursor: pointer;
  height: 1.5rem;
  min-width: 8rem;
  text-align: center;
}

.earthquake-list tr td {
  padding: 0.5rem;
  text-align: center;
}

.earthquake-list tr th:after {
  font-size: 0.6rem;
  padding-right: 0.5rem;
}

.earthquake-list .ascending:after {
  content: ' \25B2';
}

.earthquake-list .descending:after {
  content: ' \25BC';
}

.earthquake-list .primary {
  text-align: left;
}