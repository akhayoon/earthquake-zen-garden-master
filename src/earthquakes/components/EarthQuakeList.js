import React from 'react';
import { Link } from 'react-router-dom';
import formatTime from '../../common/utils/formatTime';
import earthQuakeData from '../../data/earthquakes.json';
import useSortableData from '../hooks/useSortableData';
import './EarthQuakeList.css';

export default function EarthQuakeList() {
  // Called custom memoized hook for sorting
  const { items, requestSort, sortConfig } = useSortableData(earthQuakeData.data.features);

  const getClassNamesFor = (name) => {
    if (sortConfig && sortConfig.key === name) {
      // This class gets added to show if the carat is up or down
      return sortConfig.direction === 'ascending' ? 'ascending' : 'descending';
    }
    // Return nothing for the cart by default
    return '';
  };

  return (
    <div className="container">
      <h1>USGS All Earthquakes, Past Hour</h1>
      <table className="earthquake-list">
        <thead>
          <tr>
            <th onClick={() => requestSort('place')} className={getClassNamesFor('place')}>Title</th>
            <th onClick={() => requestSort('mag')} className={getClassNamesFor('mag')}>Magnitude</th>
            <th onClick={() => requestSort('time')} className={getClassNamesFor('time')}>Time</th>
          </tr>
        </thead>
        <tbody>
        {items.map((earthquake) => {
          return (
            <tr key={earthquake.id}>
              <td data-label="Title" className="primary">
                <Link to={`/earthquakes/${earthquake.id}`}>
                  {earthquake.properties.place}</Link>
              </td>
              <td data-label="Magnitude">{earthquake.properties.mag}</td>
              <td data-label="Time">{formatTime(earthquake.properties.time)}</td>
            </tr>
          );
        })}
        </tbody>
      </table>
    </div>
  );
}
