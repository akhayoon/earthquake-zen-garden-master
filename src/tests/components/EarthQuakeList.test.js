import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
import EarthQuakeList from '../../earthquakes/components/EarthQuakeList';

describe('Test EarthQuakeList Component', () => {
  test('it should match Snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <EarthQuakeList />
        </MemoryRouter>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
