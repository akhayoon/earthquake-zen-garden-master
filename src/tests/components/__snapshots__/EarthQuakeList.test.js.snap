// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EarthQuakeList Component it should match Snapshot 1`] = `
<div
  className="container"
>
  <h1>
    USGS All Earthquakes, Past Hour
  </h1>
  <table
    className="earthquake-list"
  >
    <thead>
      <tr>
        <th
          className=""
          onClick={[Function]}
        >
          Title
        </th>
        <th
          className=""
          onClick={[Function]}
        >
          Magnitude
        </th>
        <th
          className=""
          onClick={[Function]}
        >
          Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/nc72999021"
            onClick={[Function]}
          >
            3km ENE of The Geysers, CA
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          1.19
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 19:25 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/ci38148464"
            onClick={[Function]}
          >
            11km NE of Aguanga, CA
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          0.54
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 19:24 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/ak19233683"
            onClick={[Function]}
          >
            65km W of Anchor Point, Alaska
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          1.5
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 19:18 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/nn00630289"
            onClick={[Function]}
          >
            31km SE of Bridgeport, California
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          2.3
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 19:10 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/ak19233673"
            onClick={[Function]}
          >
            13km NNW of Fritz Creek, Alaska
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          1.5
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 19:09 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/us2000dzwr"
            onClick={[Function]}
          >
            24km ESE of Nemuro, Japan
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          5.3
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 19:00 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/ci38148440"
            onClick={[Function]}
          >
            10km NE of Aguanga, CA
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          0.93
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 18:57 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/ak19233602"
            onClick={[Function]}
          >
            44km NNW of Valdez, Alaska
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          1.6
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 18:43 PM
        </td>
      </tr>
      <tr>
        <td
          className="primary"
          data-label="Title"
        >
          <a
            href="/earthquakes/ak19233598"
            onClick={[Function]}
          >
            133km WSW of Ruby, Alaska
          </a>
        </td>
        <td
          data-label="Magnitude"
        >
          1.6
        </td>
        <td
          data-label="Time"
        >
          Apr 13, 2018 18:39 PM
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
